csi/controller.go

```diff
@@ -915,7 +915,7 @@ func (s *OsdCsiServer) CreateSnapshot(
 				SnapshotId:     v.GetId(),
 				SourceVolumeId: v.GetSource().GetParent(),
 				CreationTime:   v.GetCtime(),
-				ReadyToUse:     true,
+				ReadyToUse:     v.GetError() == "",
```

Comment:
Let's also check for volume up:
`*v.Status.Enum() == api.VolumeStatus_VOLUME_STATUS_UP`

Add v.GetError() and the above check in a helper function. i.e. isSnapshotReady

Reply:
fixed in the latest commit